### PR TITLE
ISSUE-158 Manage James dependencies with a BOM

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: tmail-backend/pom.xml [hunk 1]
```diff
@@ -236,647 +195,49 @@
...
             </dependency>
             <dependency>
                 <groupId>${james.groupId}</groupId>
-                <artifactId>james-server-guice-data-ldap</artifactId>
-                <version>${james.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>org.apache.james</groupId>
-                <artifactId>james-server-guice-distributed</artifactId>
+                <artifactId>jmap-rfc-8621-integration-tests-common</artifactId>
```
Comment (Arsnael) [inline tmail-backend/pom.xml:220]: Do we still need this then?

**Review by Arsnael — COMMENTED** [review]
Can you explain a bit more how it works then? I'm confused at to what to review exactly

File: tmail-backend/pom.xml [hunk 2]
```diff
@@ -236,647 +195,49 @@
...
-                <artifactId>james-server-guice-configuration</artifactId>
-                <version>${james.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>${james.groupId}</groupId>
-                <artifactId>james-server-guice-common</artifactId>
+                <artifactId>james-project</artifactId>
                 <version>${james.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
```
Comment (chibenwa) [inline tmail-backend/pom.xml:209]: Please maven do take the deoendency management section of the pom.xml of `james-project` and put it here as if I had been copy and pasting it.

File: tmail-backend/pom.xml (hunk 1 above)
Comment (chibenwa) [reply tmail-backend/pom.xml:220]: Not defined in the BOM POM thus we need an explicit statement.

File: tmail-backend/pom.xml (hunk 2 above)
Comment (chibenwa) [reply tmail-backend/pom.xml:209]: #72laBalle

Comment (Arsnael) [reply tmail-backend/pom.xml:209]: Why need to import the BOM of james-server-guice then?I think all submodules of james-server-guice are being declared in the james-project maven file no?

File: tmail-backend/pom.xml [hunk 3]
```diff
@@ -236,647 +195,49 @@
...
             </dependency>
             <dependency>
                 <groupId>${james.groupId}</groupId>
-                <artifactId>james-server-guice-common</artifactId>
-                <type>test-jar</type>
+                <artifactId>james-server-guice</artifactId>
```
Comment (Arsnael) [inline tmail-backend/pom.xml:213]: Why need to import the BOM of james-server-guice then?I think all submodules of james-server-guice are being declared in the james-project maven file no?

Comment (chibenwa) [reply tmail-backend/pom.xml:213]: THat is what I did.

Comment (chibenwa) [reply tmail-backend/pom.xml:213]: Screenshot from 2021-05-31 10-57-24

Comment (Arsnael) [reply tmail-backend/pom.xml:213]: you did not really answered my question but ok I checked a bit myself more in details and indeed there is extra modules declared in james-server-guice...

**Review by Arsnael — APPROVED** [review]

**Pull request merged**